server/dive_utils/serializers/viame.py

```diff
@@ -233,31 +234,61 @@ def calculate_attribute_types(
             metadata_attributes[attributeKey]['datatype'] = attribute_type
 
 
-def load_csv_as_tracks_and_attributes(rows: List[str]) -> Tuple[dict, dict]:
+def load_csv_as_tracks_and_attributes(
+    rows: List[str], imageMap: Dict[str, int] = None
+) -> Tuple[dict, dict]:
     """
-    Convert VIAME CSV to json tracks.
-    Expect detections to be in increasing order (either globally or by track).
+    Convert VIAME CSV to json tracks
+
+    :param rows: string rows of a VIAME CSV file
+    :param imageMap: map of image names to frame numbers.  keys do NOT include file extension
     """
-    reader = csv.reader(row for row in rows if (not row.startswith("#") and row))
+    reader = csv.reader(row for row in rows)
     tracks: Dict[int, Track] = {}
     metadata_attributes: Dict[str, Dict[str, Any]] = {}
     test_vals: Dict[str, Dict[str, int]] = {}
+    reordered = False
+
     for row in reader:
+        if len(row) == 0 or row[0].startswith('#'):
+            # This is not a data row
+            continue
         (
             feature,
             attributes,
             track_attributes,
             confidence_pairs,
         ) = _parse_row_for_tracks(row)
 
-        trackId, _, frame, _, _ = row_info(row)
+        trackId, imageFile, _, _, _ = row_info(row)
+
+        if imageMap:
+            # validate image ordering if the imageMap is provided
+            imageName, _ = os.path.splitext(imageFile)
```

Comment:
might want to `os.path.basename(imageFile)` first.  I don't know these are typically generated from VIAME/View but the desktop version can handle paths being included in the filename specification and for upload we probably want to remove all `C:/User/` or any `/home/folder` paths and rely on the root filename.

Reply:
imageFile is a girder item name, it will never be a path.  Unless I'm wrong about where this could get called.